packages/app/src/cli/services/app/config/push.test.ts

```diff
@@ -412,4 +420,68 @@ app_preferences > url: this url is blocked 6`)
     expect(confirmPushChanges).toHaveBeenCalled()
     expect(renderSuccess).not.toHaveBeenCalled()
   })
+
+  test('does not update webhooks configurations if they already exist when declarative webhooks beta is toggled', async () => {
+    vi.mocked(fetchOrgFromId).mockResolvedValue(orgWithDeclarativeWebhookBeta)
+
+    // app already exists with the following config
+    const app = testAppWithConfig({
+      config: {
+        ...EXISTING_APP_CONFIG,
+        gdprWebhooks: {
+          customerDataRequestUrl: 'https://test.com',
+          customerDeletionUrl: undefined,
+          shopDeletionUrl: undefined,
+        },
+      },
+    })
+
+    // user tries to push up new webhook api version
+    app.configuration = {
+      ...app.configuration,
+      webhooks: {
+        api_version: '2023-01',
+      },
+    }
+    const options: PushOptions = {
+      configuration: app.configuration,
+      force: true,
+    }
+
+    vi.mocked(partnersRequest).mockResolvedValue({
+      app: {
+        apiKey: '12345',
+        webhookApiVersion: '2023-04',
```

Comment:
this should remain the same version as the `EXISTING_APP_CONFIG` because it skipped the `2023-01` webhookApiVersion we were trying to push up above